### ISSUE TITLE
[Table] [A11y] headers="expand" should be removed from first column <td>

### ISSUE THREAD

**Issue by valZho** [carbon-design-system/carbon-addons-iot-react#3485]
### What package is this for?

- [X] React
- [?] Angular

### Describe the bug

The 'headers' attribute with value "expand" does not reference a valid 'id' in the document

### Expected behavior

The <td> tags for the first column (row expansion) in a table contain the property `header="expand"` which is supposed to point to the id of a <th> cell, but none of the <th> elements have ids (let alone something so generic). The header property should be removed.

### Environment/versions:

- `carbon-addons-iot-react` version: 2.150.0

### Additional context

![Screen Shot 2022-06-23 at 3 16 44 PM](https://user-images.githubusercontent.com/294863/175391852-84d0db80-7567-4c01-b2e0-fb262ad68d84.png)

**Comment by davidicus** [comment]
This attribute is coming from Carbon. After looking into [their own issue ](https://github.com/carbon-design-system/carbon/issues/10920#issuecomment-1060898367)for this the fix is to add id to their TableHeader component as they have done in this [PR](https://github.com/carbon-design-system/carbon/pull/10547)

**Comment by jessieyan** [comment]
On next branch, I can still see headers='expand' on the TableExpandRow with no id. But I don't see any violation that shown before. 

![image](https://user-images.githubusercontent.com/24279794/196779133-55ac86b7-3147-42b5-8819-b5c426e95fac.png)

![image](https://user-images.githubusercontent.com/24279794/196779230-ea992f2d-6fef-4a89-bc65-2479cb5f4dd7.png)

@davidicus Should we close this as fixed?
